src/Internal.AspNetCore.Sdk/build/Git.targets

```diff
@@ -19,6 +21,29 @@
     </Sdk_GetGitCommitInfo>
   </Target>
 
+  <Target Name="CreateSourceLink"
+          Condition="'$(SourceLink)'=='' AND '$(GenerateSourceLinkFile)' != 'false' "
+          Inputs="$(IntermediateCommitHash);$(MSBuildAllProjects)"
+          Outputs="$(SourceLinkDestination)" >
+
+    <PropertyGroup>
+      <IsSourceProject>$(MSBuildProjectDirectory.StartsWith($(SourceDirectory)))</IsSourceProject>
+    </PropertyGroup>
+
+    <Sdk_CreateSourceLink
+          RootDirectory="$(RepositoryRoot)"
+          OriginUrl="$(RepositoryUrl)"
+          Commit="$(CommitHash)"
+          DestinationFile="$(SourceLinkDestination)"
+          ContinueOnError="WarnAndContinue"
+          Condition="'$(CommitHash)' != '' AND '$(RepositoryUrl)' != '' AND '$(IsSourceProject)' == 'True' ">
+      <Output TaskParameter="SourceLinkFile" PropertyName="SourceLink" />
+    </Sdk_CreateSourceLink>
+
+    <Warning Text="SourceLink not enabled because this is not a git repo."
```

Comment:
Also add a warning if `RepositoryUrl` is empty. We should be setting this everywhere sourcelink it used.